### ISSUE TITLE
W3C Schema parser broken for most facets and constraints

### ISSUE THREAD

**Issue by GoogleCodeExporter** [TIBCOSoftware/genxdm#63]
```
As a part of removing the <A> parameter from schema model, the Schema Parser's 
use of AtomBridge was removed.  The schema parser implementation used the atom 
bridge to validate the values of various constraints.

First: it is questionable whether this is the right approach.  A supplied atom 
bridge may not behave correctly.  Should the behavior of the schema parser be 
that variable and caller-defined?  I doubt that that was the intent.

Second: it can be fixed by creating a canonical atom bridge, used only by the 
schema parser.  This may be the least extensive change to make.

Third: it can be fixed by implementing the validation functionality of atom 
bridge types without having an atom bridge.

Fourth: the biggest problem is schema constraint checker, which is a procedural 
library for validating schema.  As designs go, procedural code in java is 
perhaps my least favorite.

This issue is marked blocked by 13.  It only exists if the solution to 13 is 
accepted, and merged into trunk.  At that point, it becomes rather high 
priority.
```

Original issue reported on code.google.com by `aale...@gmail.com` on 6 Jun 2011 at 3:28
- Blocking: #64
- Blocked on: #13
